File: core/src/main/java/io/github/zncmn/libyuv/YuvConvert.java
package io.github.zncmn.libyuv;

import org.jetbrains.annotations.NotNull;

public class YuvConvert {
    public void fromARGB(@NotNull final byte[] argb, @NotNull final byte[] yuv, final int width, final int height, @NotNull final YuvFormat format) {
        if (format == YuvFormat.NV21) {
            Yuv.INSTANCE.abgrToNv21(argb, yuv, width, height);
        } else {
            throw new UnsupportedOperationException("not supported format. " + format);
        }
    }

    public void toARGB(@NotNull final byte[] yuv, @NotNull final byte[] argb, final int width, final int height, @NotNull final YuvFormat format) {
        if (format == YuvFormat.NV21) {
            Yuv.INSTANCE.nv21ToAbgr(yuv, argb, width, height);
        } else {
            throw new UnsupportedOperationException("not supported format. " + format);
        }
    }
}

File: core/src/main/java/io/github/zncmn/libyuv/YuvConvert.kt
package io.github.zncmn.libyuv

class YuvConvert {
    init {
        System.loadLibrary("yuv")
    }

    fun argbToYuv(rgb: ByteArray, width: Int, height: Int, yuv: ByteArray) {
        val bgr = ByteArray(rgb.size)
        abgrToArgb(rgb, width, height, bgr)
        argbToNv21(bgr, width, height, yuv)
    }

    private external fun abgrToArgb(rgb: ByteArray, width: Int, height: Int, bgr: ByteArray)
    private external fun argbToNv21(bgr: ByteArray, width: Int, height: Int, yuv: ByteArray)
}
